app/main.py

```diff
@@ -1,213 +1,33 @@
-        if "power" in arthur["potion"]["effect"]:
-            arthur["power"] += arthur["potion"]["effect"]["power"]
-
-        if "protection" in arthur["potion"]["effect"]:
-            arthur["protection"] += arthur["potion"]["effect"]["protection"]
-
-        if "hp" in arthur["potion"]["effect"]:
-            arthur["hp"] += arthur["potion"]["effect"]["hp"]
-
-    # mordred
-    mordred = knightsConfig["mordred"]
-
-    # apply armour
-    mordred["protection"] = 0
-    for a in mordred["armour"]:
-        mordred["protection"] += a["protection"]
-
-    # apply weapon
-    mordred["power"] += mordred["weapon"]["power"]
-
-    # apply potion if exist
-    if mordred["potion"] is not None:
-        if "power" in mordred["potion"]["effect"]:
-            mordred["power"] += mordred["potion"]["effect"]["power"]
-
-        if "protection" in mordred["potion"]["effect"]:
-            mordred["protection"] += mordred["potion"]["effect"]["protection"]
-
-        if "hp" in mordred["potion"]["effect"]:
-            mordred["hp"] += mordred["potion"]["effect"]["hp"]
-
-    # red_knight
-    red_knight = knightsConfig["red_knight"]
-
-    # apply armour
-    red_knight["protection"] = 0
-    for a in red_knight["armour"]:
-        red_knight["protection"] += a["protection"]
-
-    # apply weapon
-    red_knight["power"] += red_knight["weapon"]["power"]
-
-    # apply potion if exist
-    if red_knight["potion"] is not None:
-        if "power" in red_knight["potion"]["effect"]:
-            red_knight["power"] += red_knight["potion"]["effect"]["power"]
-
-        if "protection" in red_knight["potion"]["effect"]:
-            red_knight["protection"] += red_knight["potion"]["effect"]["protection"]
-
-        if "hp" in red_knight["potion"]["effect"]:
-            red_knight["hp"] += red_knight["potion"]["effect"]["hp"]
-
-    # -------------------------------------------------------------------------------
-    # BATTLE:
+from app.for_act.knights import KNIGHTS
+from app.for_act.defeat import defeat_check
+from app.for_act.preparation import prepare_to_the_battle
 
-    # 1 Lancelot vs Mordred:
-    lancelot["hp"] -= mordred["power"] - lancelot["protection"]
-    mordred["hp"] -= lancelot["power"] - mordred["protection"]
 
-    # check if someone fell in battle
-    if lancelot["hp"] <= 0:
-        lancelot["hp"] = 0
+def battle(knights_config: dict) -> dict:
 
-    if mordred["hp"] <= 0:
-        mordred["hp"] = 0
+    knight_list = []
+    for name in knights_config:
+        knight = knights_config[name]
+        prepare_to_the_battle(knight)
+        knight_list.append(name)
 
-    # 2 Arthur vs Red Knight:
-    arthur["hp"] -= red_knight["power"] - arthur["protection"]
-    red_knight["hp"] -= arthur["power"] - red_knight["protection"]
+    knight1 = knights_config[knight_list[0]]
+    knight2 = knights_config[knight_list[1]]
+    knight3 = knights_config[knight_list[2]]
+    knight4 = knights_config[knight_list[3]]
```

Comment:
try to make it with loop

Reply:
I'm sorry I can't. Could you, please, help me? I don't know how to create variables with new names in a loop, something like this:
for i in range(4):
knight_i = knights_config[knight_list[i]] :(

Reply:
Look at this link:
https://stackoverflow.com/questions/6181935/how-do-you-create-different-variable-names-while-in-a-loop. 

There are a lot of different ways how to do it, but I think the better is python dict. You can use a knight name like a key, and write key and value in the loop.

Reply:
UPD: I solved the loop problem with:
    for i in range(4):
        globals()["knight" + str(i + 1)] = knights_config[knight_list[i]]

and now I can't agree the next lines for flake8 mistakes: [.\app\main.py:17:5: F821 undefined name 'knight1' 
etc.]

Reply:
> Look at this link: https://stackoverflow.com/questions/6181935/how-do-you-create-different-variable-names-while-in-a-loop.
> 
> There are a lot of different ways how to do it, but I think the better is python dict. You can use a knight name like a key, and write key and value in the loop.

Thank you very much! Now with a dict, it's a single line:)